File: code-generator/src/main/java/io/github/potjerodekool/openapi/internal/di/conditiontest/ConditionalOnDependencyTest.java
package io.github.potjerodekool.openapi.internal.di.conditiontest;

import io.github.potjerodekool.openapi.common.dependency.DependencyChecker;
import io.github.potjerodekool.openapi.internal.di.bean.BeanDefinition;
import io.github.potjerodekool.openapi.common.dependency.ConditionalOnDependency;

public class ConditionalOnDependencyTest implements ConditionalTest<ConditionalOnDependency> {

    private final DependencyChecker dependencyChecker;

    public ConditionalOnDependencyTest(final DependencyChecker dependencyChecker) {
        this.dependencyChecker = dependencyChecker;
    }

    @Override
    public boolean test(final ConditionalOnDependency conditionalOnDependency,
                        final BeanDefinition beanDefinition) {
        final var groupId = conditionalOnDependency.groupId();
        final var artifactId = conditionalOnDependency.artifactId();
        return this.dependencyChecker.isDependencyPresent(groupId, artifactId);
    }
}

File: code-generator/src/main/kotlin/io/github/potjerodekool/openapi/internal/di/conditiontest/ConditionalOnDependencyTest.kt
package io.github.potjerodekool.openapi.internal.di.conditiontest

import io.github.potjerodekool.openapi.common.dependency.ConditionalOnDependency
import io.github.potjerodekool.openapi.common.dependency.DependencyChecker
import io.github.potjerodekool.openapi.internal.di.bean.BeanDefinition

class ConditionalOnDependencyTest(private val dependencyChecker: DependencyChecker) :
    ConditionalTest<ConditionalOnDependency> {
    override fun test(condition: ConditionalOnDependency, beanDefinition: BeanDefinition): Boolean {
        val groupId = condition.groupId
        val artifactId = condition.artifactId
        return dependencyChecker.isDependencyPresent(groupId, artifactId)
    }
}
